### DISCUSSION TITLE
How can I add more languages to the website?

### DISCUSSION THREAD

**Discussion by Najib90** [mearashadowfax/ScrewFast#214] [Q&A] [answered]
Currently, the website supports both French and English languages. I’m interested in expanding the multilingual capabilities by adding languages such as German and Dutch. Could you please provide guidance on the following:

**Files and Directories**: Which specific files or directories should I modify or add to implement the new languages?
**Configuration Settings**: Are there any configuration files or settings that need to be updated to support additional languages?
**Translation Management**: What is the best practice for managing translations within the current project structure?
**Dependencies**: Are there any dependencies or libraries required for adding new language support?

Any examples, documentation links, or step-by-step instructions would be greatly appreciated.

**Comment by mearashadowfax** [comment, marked as answer]
Hey @Najib90! You have a few options:

**Content Collections:** You can continue using Content Collections to store localized content for different languages and query them as needed. For example, create or adjust the Hero component by setting up a collection directory called `hero`. Then, add JSON files for each language, such as `heroEN.json`, `heroFR.json`, and for the new languages, `heroDE.json` and `heroNL.json`.

**CMS Integration:** Another option is to use a CMS to manage your content, which follows similar logic to Astro's native Content Collections. The content will be stored centrally and queried based on the active language.

**i18n Library for Shared Components:** While Content Collections are great for page-specific content, you’ll still want an i18n library for translating strings in components shared across multiple pages (e.g., Layouts or reusable components). I recommend checking out [Paraglide-Astro](https://inlang.com/m/iljlwzfs/paraglide-astro-i18n). It integrates smoothly with Astro's native i18n handling for language detection, so you won't need to duplicate logic. Plus, messages are fully typed and treeshaken, ensuring only necessary translations are sent to the client.

Useful links:
https://docs.astro.build/en/guides/internationalization/
https://github.com/jlarmstrongiv/astro-i18n-aut
https://github.com/LorisSigrist/paraglide-astro-example
